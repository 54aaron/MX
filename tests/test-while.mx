int main() {
    int a;
    int b;
    int c;
    a = 0;
    b = 3;
    c = 5;

    prints("==============================================");
    prints("while( a < 10)");
    while(a < 10) {
        if( a == 5) {
            prints("a is equal to 5");
        } else {
            print(a);
        }
        a = a + 1;
    }

    prints("==============================================");
    prints("c = 5 and b = 3");
    prints("while( c > b)");
    while( c > b) {
        print(c);
        c = c - 1;
    }

}
