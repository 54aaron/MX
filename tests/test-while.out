==============================================
while( a < 10)
0
1
2
3
4
a is equal to 5
6
7
8
9
==============================================
c = 5 and b = 3
while( c > b)
5
4
